app/uk/gov/hmrc/securemessage/services/ImplicitClassesExtensions.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.securemessage.services
+
+import uk.gov.hmrc.auth.core.Enrolments
+import uk.gov.hmrc.securemessage.ParticipantNotFound
+import uk.gov.hmrc.securemessage.models.core.{ Conversation, CustomerEnrolment, Identifier, Participant }
+
+trait ImplicitClassesExtensions {
+  implicit class EnrolmentsExtensions(enrolments: Enrolments) {
```

Comment:
These class conversions are not  extensions but implicit conversions.The good practice is to avoid implicit converstions.
https://contributors.scala-lang.org/t/can-we-wean-scala-off-implicit-conversions/4388

Reply:
this is not implicit conversion, implicit conversion is an implicit function, this is an implicit class which is precisely what is it being called an extension, it is the Scala way of implementing the decorator pattern :)
Implicit conversions are bad cause they are kind of magin, implicit classes are ok, they allow extending a class with useful functions.